02-microarray/differential-expression_microarray_02_several-groups.Rmd

```diff
@@ -0,0 +1,454 @@
+  dplyr::filter(dplyr::n() > 3) %>%
+  dplyr::ungroup()
+```
+
+Let's take a look at the subgroup summary again. 
+
+```{r}
+summary(as.factor(filtered_metadata$subgroup))
+```
+
+Note that the `U` and the `SHH OUTLIER` samples are gone and only the four groups we are interested in are left. 
+
+But, we still need to filter these samples out from the expression data that's stored in `df`. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(filtered_metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), filtered_metadata$geo_accession)
+```
+
+## Create the design matrix
+
+`limma` needs a numeric design matrix to signify which samples are of which subtype of medulloblastoma. 
+Now we will create a model matrix based on our `subgroup` variable.
+We are using a `+ 0` in the model so no particular group is treated as the control. 
+
+```{r}
+# Create the design matrix
+des_mat <- model.matrix(~ filtered_metadata$subgroup + 0)
+```
+
+Let's take a look at what our design matrix looks like. 
+
+```{r}
+# Print out the design matrix
+head(des_mat)
+```
+
+The design matrix column names are a bit messy, so we will neaten them up by dropping the `filtered_metadata$subgroup` designation they all have. 
+
+```{r}
+# Make the column names less messy
+colnames(des_mat) <- stringr::str_remove(colnames(des_mat), "filtered_metadata\\$subgroup")
+```
+
+Side note: If you are wondering why there are added two `\` above in `"filtered_metadata\\$subgroup"`, that's called an [escape character](https://cran.r-project.org/web/packages/stringr/vignettes/regular-expressions.html#escaping). 
+There's a whole universe of things called [regular expressions (regex)](https://cran.r-project.org/web/packages/stringr/vignettes/regular-expressions.html) that can be super handy for string manipulations. 
+
+## Perform differential expression
+
+Now we are ready to actually start our setting up our differential expression model. 
+To accommodate our design that [has more than 2 groups this time](), we will need to do this in a couple steps. 
+
+First we need to set up our basic linear model then apply empirical Bayes smoothing.  
+
+```{r}
+# Apply linear model to data
+fit <- lmFit(df, design = des_mat)
+
+# Apply empirical Bayes to smooth standard errors
+fit <- eBayes(fit)
+```
+
+Now that we have our basic model fitting, we need to make the contrasts between all our groups. 
+Depending on your scientific questions, you will need to switch this around.
+Consulting the [limma users guide](https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf) for how to alter these steps is good idea. 
+
+In this contrasts matrix, we are comparing each subtype to all the other subtypes.  
+We're diving by three in this expression so that each is contrasted to the average of the other three groups (`makeContrasts` doesn't allow you to use functions like `mean()`, it wants a formula). 
```

Comment:
```suggestion
We're dividing by three in this expression so that each is contrasted to the average of the other three groups (`makeContrasts` doesn't allow you to use functions like `mean()`, it wants a formula). 
```

Reply:
🤿